addons/point_of_sale/models/pos_session.py

```diff
@@ -379,14 +384,20 @@ def _close_session_action(self, amount_to_balance):
             'context': {**self.env.context, 'active_ids': self.ids, 'active_model': 'pos.session'},
         }
 
-    def close_session_from_ui(self):
+    def close_session_from_ui(self, other_payment_method_diff_pairs=None):
         """Calling this method will try to close the session.
 
         If successful, it returns {'successful': True}
         Otherwise, it returns {'successful': False, 'message': str, 'redirect': bool}.
         'redirect' is a boolean used to know whether we redirect the user to the back end or not.
         When necessary, error (i.e. UserError, AccessError) is raised which should redirect the user to the back end.
         """
+
+        other_payment_method_diff_pairs = other_payment_method_diff_pairs or []
+        other_payment_method_diffs = {}
+        for payment_method_id, diff_amount in other_payment_method_diff_pairs:
```

Comment:
if other_payment_method_diff_pairs is a list of tuple (pm_id,diff). You can just do dict(other_payment_method_diff_pairs), and it'll trasform it to a dictionnary